### PR TITLE
feat: associate files to be opened in cpeditor

### LINKED ISSUE #855: Associate source files to open in cpeditor
#846 and #849 only do this for macOS, I think we should do it for all OSs.

### PR DESCRIPTION
**Pull request by swift-zym**
## Description

Update desktop file(Linux) and Inno Setup file(Windows) to associate files to open in cpeditor.
## Related Issues / Pull Requests

Closes #855
## How Has This Been Tested?

https://github.com/swift-zym/cpeditor/releases/tag/0.0.8
## Screenshots (if appropriate)
image

## Checklist

- [x] If the key of a setting is changed, the `old` attribute is updated or it is resolved in SettingsUpdater.
- [x] If there are changes of the text displayed in the UI, they are wrapped in `tr()` or `QCoreApplication::translate()`.
- [x] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [x] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by swift-zym** [comment]
I don't have a windows device by now so I can't test it, can you help me to test it @ouuan ?

File: dist/linux/cpeditor.desktop [hunk 1]
```diff
@@ -14,3 +14,5 @@ Terminal=false
 StartupNotify=true
 Encoding=UTF-8
 Categories=Development;TextEditor;Qt;
+MimeType=text/plain;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-c;text/x-c++;
```
Comment (ouuan) [inline dist/linux/cpeditor.desktop:17]:
```suggestion
MimeType=text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-python
```

Comment (swift-zym) [reply dist/linux/cpeditor.desktop:17]: Some OSs may mark code files as `text/plain`.

Comment (ouuan) [reply dist/linux/cpeditor.desktop:17]: > Some OSs may mark code files as `text/plain`.

Can you give a reference? And I also want to know where did you find `text/x-c` and `text/x-c++` :thinking:

**Comment by swift-zym** [comment]
It seems work well on Windows and Linux now.

**Comment by ouuan** [comment]
I'm not sure whether it works on Windows correctly. I don't have a clean Windows environment, so I tried to clean my environment and it didn't seem to work. I'm not sure whether it's caused by my incorrect way to clean the environment.

@swift-zym Did you test it in a clean Windows environment?

**Comment by swift-zym** [comment]
> I'm not sure whether it works on Windows correctly. I don't have a clean Windows environment, so I tried to clean my environment and it didn't seem to work. I'm not sure whether it's caused by my incorrect way to clean the environment.
> 
> @swift-zym Did you test it in a clean Windows environment?

I didn't, I find the problem now...

**Comment by swift-zym** [comment]
Now I've test it in a clean Windows environment and it works well.

**Review by ouuan — APPROVED** [review]
Haven't tested on my Windows, but the diff looks good and @swift-zym has tested it.

**Comment by coder3101** [comment]
Can you provide a installer build from this branch, so that I can test and approve. The URL above gives 404

**Comment by swift-zym** [comment]
> Can you provide a installer build from this branch, so that I can test and approve. The URL above gives 404

https://github.com/swift-zym/cpeditor/releases/tag/0.0.8

**Review by coder3101 — APPROVED** [review]
Works fine!

**Pull request merged**